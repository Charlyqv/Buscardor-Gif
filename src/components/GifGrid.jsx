import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category ) ;

  return (
    <>
      <h3>{ category }</h3>
      {
        isLoading && ( <h2>Cargando...</h2> )
      }
      

      <div className='card-grid'>
        {
          images.map( ( image ) => (
            <GifItem 
              key={ image.id }
              {...image}
              // La linea de arriba es para obtener las propiedades del elemento, es lo mismo 
              // que la las dos lineas de abajo
              // tittle={ image.tittle }
              // url={ image.url }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}