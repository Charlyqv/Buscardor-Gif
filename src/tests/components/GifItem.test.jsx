import { render } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en GifItem', () => {

  const title = 'Pikachu';
  const url = 'https://pikachu.com/pikapika.jpg';

  test('Debe hacer match con el snapshot', () => {

    const { container } = render( <GifItem title={ title } url={ url } /> );
    expect( container).toMatchSnapshot();

  });
});