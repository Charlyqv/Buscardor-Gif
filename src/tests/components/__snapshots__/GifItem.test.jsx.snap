// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pikachu"
      src="https://pikachu.com/pikapika.jpg"
    />
    <p>
      Pikachu
    </p>
  </div>
</div>
`;
